tests/phpunit/Presentation/UiBuilderTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Tests\Presentation;
+
+use PHPUnit\Framework\TestCase;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\Config;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\PropertyConstraints;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\PropertyConstraintsList;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\Query;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\QueryStringParser;
+use ProfessionalWiki\WikibaseFacetedSearch\Presentation\UiBuilder;
+use ProfessionalWiki\WikibaseFacetedSearch\Tests\TestDoubles\SpyFacetHtmlBuilder;
+use ProfessionalWiki\WikibaseFacetedSearch\Tests\TestDoubles\SpyTemplateParser;
+use ProfessionalWiki\WikibaseFacetedSearch\Tests\TestDoubles\StubQueryStringParser;
+use ProfessionalWiki\WikibaseFacetedSearch\WikibaseFacetedSearchExtension;
+use RuntimeException;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\NumericPropertyId;
+
+/**
+ * @covers \ProfessionalWiki\WikibaseFacetedSearch\Presentation\UiBuilder
+ * @covers \ProfessionalWiki\WikibaseFacetedSearch\WikibaseFacetedSearchExtension
+ */
+class UiBuilderTest extends TestCase {
+
+	public function testIntegrationSmoke(): void {
+		try {
+			WikibaseFacetedSearchExtension::getInstance()->getConfig()->getInstanceOfId();
+		} catch ( RuntimeException ) {
+			$this->markTestSkipped( 'No valid config available' );
+		}
+
+		$html = WikibaseFacetedSearchExtension::getInstance()->getUiBuilder()->createHtml( 'foo' );
+		$this->assertStringContainsString( 'topbar', $html );
+		$this->assertStringContainsString( 'sidebar', $html );
+	}
+
+	public function testTabsViewModelContainsItemTypeProperty(): void {
+		$config = new Config( instanceOfId: new NumericPropertyId( 'P1337' ) );
+		$templatePsy = new SpyTemplateParser();
+
+		$this->newUiBuilder( config: $config, templatePsy: $templatePsy )
+			->createHtml( 'unimportant' );
+
+		$this->assertSame(
+			'P1337',
+			$templatePsy->getArgs()['instanceId']
+		);
+	}
+
+	private function newUiBuilder(
+		?Config $config = null,
+		?SpyTemplateParser $templatePsy = null,
+		?QueryStringParser $queryStringParser = null
+	): UiBuilder {
+		return new UiBuilder(
+			$config ?? new Config(),
+			new SpyFacetHtmlBuilder(),
+			$templatePsy ?? new SpyTemplateParser(),
+			$queryStringParser ?? new StubQueryStringParser()
+		);
+	}
+
+	public function testTabsViewModelContainsItemTypes(): void {
+		$config = WikibaseFacetedSearchExtension::getInstance()->newConfigDeserializer()->deserialize( <<<JSON
```

Comment:
I went with deserializing JSON here to avoid binding to unrelated implementation details of Config, and because those implementation details have issues as per https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/107